app/components/reporting/reporting-roles-edit/reporting-roles-edit-directive.js

```diff
@@ -58,9 +62,40 @@ angular.module('roles.edit.directive', [
           return false;
         };
 
+        $scope.getDirtyRates = function () {
+          var dirty = [];
+          // // Validate if flat_rate is dirty
+          if ($scope.model.hasOwnProperty('flat_rate') && !_.isEmpty($scope.model.flat_rate.$dirty())) {
+            dirty.push($scope.model.flat_rate);
+          }
+
+          // Validate if hourly_rate is dirty
+          if ($scope.model.hasOwnProperty('hourly_rate')) {
+            if (!_.isEmpty($scope.model.hourly_rate.$dirty())) {
+              dirty.push($scope.model.hourly_rate);
+            }
+          }
+
+          // Validate if feature_type_rates are dirty
+          for (var i = 0; i<$scope.model.feature_type_rates.length; i++) {
```

Comment:
I've been shifting to using `map`, `forEach`, `reduce`, etc. instead of using `for` loops. It's a bit cleaner looking. This block would become:
```javascript
$scope.model.feature_type_rates.forEach(function (rate) {
  if (!_.isEmpty(rate.$dirty()) {
    dirty.push(rate);
  }
});
```